### ISSUE TITLE
After disabling JDBC config source, datasource properties remain required

### ISSUE THREAD

**Issue by Blarc** [quarkiverse/quarkus-config-extensions#169]
I am using quarkus `3.1.1.Final` and quarkus config jdbc `2.2.0`.

I have added the JDBC config source like this:
```yaml
quarkus:
  config:
    source:
      jdbc:
        table: PARAMETER
        key: CODE
        value: VALUE_
```
I then add datasource properties to `.env` file:
```env
QUARKUS_DATASOURCE_JDBC_URL=jdbc:...
QUARKUS_DATASOURCE_USERNAME=user123
QUARKUS_DATASOURCE_PASSWORD=pass123
```
This works great, but for building the application in CI/CD pipeline I don't have a database and I don't need the properties to be fetched from the database. That's why I set the environment variable:
```env
QUARKUS_CONFIG_SOURCE_JDBC_ENABLED=false
```
However, the build fails with:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.1.1.Final:build (default) on project maas-core: Failed to build quarkus application: Configuration validation failed:
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.jdbc.url in property quarkus.config.source.jdbc.url
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.password in property quarkus.config.source.jdbc.password
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.username in property quarkus.config.source.jdbc.username
```
If I add the following environment variables, the build passes:
```env
QUARKUS_CONFIG_SOURCE_JDBC_URL=""
QUARKUS_CONFIG_SOURCE_JDBC_USERNAME=""
QUARKUS_CONFIG_SOURCE_JDBC_PASSWORD=""
```

Would it be possible to not expand the datasource properties, if `QUARKUS_CONFIG_SOURCE_JDBC_ENABLED` is set to `false` ?  You don't need those properties, if you don't use the JDBC config source.

**Comment by radcortez** [comment]
I'll have a look.

**Comment by Blarc** [comment]
I have now figured out, that these properties are used in the build process and can not be changed with environment variables after the build process. Or am I wrong?

What I did was:
1. set the following environment variables (only for the build process)
```
QUARKUS_CONFIG_SOURCE_JDBC_ENABLED=false
QUARKUS_CONFIG_SOURCE_JDBC_URL=""
QUARKUS_CONFIG_SOURCE_JDBC_USERNAME=""
QUARKUS_CONFIG_SOURCE_JDBC_PASSWORD=""
```
2. build the application by running `mvn package`
3. build a Docker image by copying the built application
4. set the following environment variables in the deployment's environment (where previous variables weren't applied)
```
QUARKUS_DATASOURCE_JDBC_URL=database_url
QUARKUS_DATASOURCE_USERNAME=database_username
QUARKUS_DATASOURCE_PASSWORD=database_password
```
5. deploy the built Docker image

I get an error, that some properties are not set, even though they are set in the database.
